altair/utils/schemapi.py

```diff
@@ -181,19 +240,53 @@ def __str__(self):
             for val in schema_path[:-1]
             if val not in (0, "properties", "additionalProperties", "patternProperties")
         )
-        message = self.message
-        if self._additional_errors:
-            message += "\n        " + "\n        ".join(
-                [e.message for e in self._additional_errors]
-            )
-        return """Invalid specification
 
-        {}, validating {!r}
+        # Output all existing parameters when an unknown parameter is specified
+        if (
+            hasattr(vegalite, schema_path.split(".")[-1])
+            and self.validator == "additionalProperties"
+        ):
+            altair_class = schema_path.split(".")[-1]
+            vegalite_core_class = getattr(vegalite, schema_path.split(".")[-1])
+            param_dict_keys = inspect.signature(vegalite_core_class).parameters.keys()
+            param_names_table = self._format_params_as_table(param_dict_keys)
+
+            # `cleandoc` removes multiline string indentation in the output
+            return inspect.cleandoc(
+                """`{}` has no parameter named {!r}
+
+                Existing parameter names are:
+                {}
+                See the help for `{}` to read the full description of these parameters
+                """.format(
+                    altair_class,
+                    self.message.split("('")[-1].split("'")[0],
+                    param_names_table,
+                    altair_class,
+                )
+            )
+        # Use the default error message for all other cases than unknown parameter errors
+        else:
+            message = self.message
+            # Add a summary line when parameters are passed an invalid value
+            # For example: "'asdf' is an invalid value for `stack`
+            if hasattr(vegalite, schema_path.split(".")[-1]) and self.absolute_path:
+                # The indentation here must match that of `cleandoc` below
+                message = f"""'{self.instance}' is an invalid value for `{self.absolute_path[-1]}`:
```

Comment:
Any idea why flake8 does not raise an error for this line? It is too long. Same for a few lines in the tests.

Reply:
I htink both black and flake8 allow multiline strings are allowed to be longer than regular strings:

![image](https://user-images.githubusercontent.com/4560057/219981662-d9266d66-801a-48b9-a25d-a15387219f56.png)

I would rather not break this line up to facilitate the formatting of the output.

Reply:
Good to know, thanks! Agree, let's keep it like it is.